src/services/auth.service.js

```diff
@@ -40,10 +41,22 @@ let authService = () => {
     }).exec()
   }
 
+  let authorize = roles => {
+    return (req, res, next) => {
+      if (!roles.includes(req.user.roles)) { //TODO: Check if a user could have multiple roles.
+        let err = new Error('Your user does not have the required role(s) to execute this action.')
+        err.status = HttpStatus.UNAUTHORIZED
+        throw err
+      }
+      next()
```

Comment:
It *IS* a middleware function. Calling `next()` is required to go to the next middleware function. As you already noticed, we're calling the `authorize()` middleware *before* calling the controller's function.